dataclients/kubernetes/kube.go

```diff
@@ -351,13 +351,23 @@ func ParsePathMode(s string) (PathMode, error) {
 	}
 }
 
-func mapRoutes(r []*eskip.Route) map[string]*eskip.Route {
+func mapRoutes(r []*eskip.Route) (map[string]*eskip.Route, []*eskip.Route) {
+	var uniqueRoutes []*eskip.Route
 	m := make(map[string]*eskip.Route)
 	for _, ri := range r {
-		m[ri.Id] = ri
+		if existingRoute, ok := m[ri.Id]; ok {
+			if eskip.Eq(existingRoute, ri) {
+				log.Warnf("Duplicate routeID found: %s", ri.Id)
+			} else {
+				log.Errorf("Duplicate routeID with different services found: %s, %s is omitted.", ri.Id, ri.String())
```

Comment:
Error message should be consistent: "Failed to do ..."
Then I would think it might make sense to tell that two route X and Y (both route.Id are interesting) are found that differ in `cmp.Diff`.